Lib/_pyrepl/console.py

```diff
@@ -160,6 +161,14 @@ def __init__(
     ) -> None:
         super().__init__(locals=locals, filename=filename, local_exit=local_exit)  # type: ignore[call-arg]
         self.can_colorize = _colorize.can_colorize()
+        self.barry_as_FLUFL = False
+
+    def check_barry_as_FLUFL(self, tree):
+        for node in ast.walk(tree):
+            if isinstance(node, ast.ImportFrom) and node.module == "__future__":
+                if any(alias.name == "barry_as_FLUFL" for alias in node.names):
+                    self.compile.compiler.flags |= getattr(__future__, "barry_as_FLUFL").compiler_flag
+                    self.barry_as_FLUFL = True
```

Comment:
```suggestion
		if self.barry_as_FLUFL:
			return

        for node in ast.walk(tree):
            if isinstance(node, ast.ImportFrom) and node.module == "__future__":
                if any(alias.name == "barry_as_FLUFL" for alias in node.names):
                    self.compile.compiler.flags |= getattr(__future__, "barry_as_FLUFL").compiler_flag
                    self.barry_as_FLUFL = True
                  	break
```

For the detection, you should have a separate routine which only retrieves the top-level future statements if any. Just walking the AST tree wouldn't be sufficient.

Reply:
Maybe you could even use `_barry_as_FLUFL` instead of `barry_as_FLUFL` (I don't think you want the property to be exposed).